docs/2-Interviews/preparing_for_interviews.md

```diff
@@ -12,22 +12,64 @@ Be prepared to:
 - talk about and work towards writing an algorithm to solve a problem, during the interview
 - you can also expect a “tell me about yourself” question (so be prepared!) and some time where they tell you about their company
 
+### What material should I know?
+
+This is a difficult question to answer. The more you know, the fewer the number of weak points that could trip you up. Knowing more things is a good thing in general, you shouldn't learn things just for interviews.
+
+That being said, you have limited time to learn things so you probably want the biggest bang for the buck.
+
+Things that interviewers will expect you to know:
+- Understanding O() analysis
+- Arrays vs linked-list
+- Stacks and queues
+- Dictionaries (one of the most used data structures in interviews)
+- Being able to traverse a binary search tree recursively.
+
+Things that are quite useful:
+- Be able to give a 5 minute answer the question "how does the internet work?" or "what happens when I type google.com in the URL bar and press enter?"
```

Comment:
Good idea. I'll look for a concise one. But tl;dr the things to know is "packets get sent around a network of routers", "you send a packet to a DNS asking for the IP address of the website you're trying to get to", "when you have the IP address you can find the server" and "the server sends you stuff". Then you can go into arbitrary depth at different steps of the process but that's usually the gist of what employers want to hear.

(I've been asked this question just once but it is very useful to know in general so I feel comfortable including it)